src/pke/lib/scheme/bfvrns/bfvrns-leveledshe.cpp

```diff
@@ -130,8 +130,10 @@ uint32_t FindLevelsToDrop(usint multiplicativeDepth, std::shared_ptr<CryptoParam
     auto noiseKS = [&](uint32_t n, double logqPrev, double w) -> double {
         if (scalTechnique == HYBRID)
             return k * (numPartQ * delta(n) * Berr + delta(n) * Bkey + 1.0) / 2;
-        else
-            return delta(n) * (floor(logqPrev / (log(2) * dcrtBits)) + 1) * w * Berr;
+        else {
+            double numDigitsPerTower = relinWindow == 0 ? 1 : floor(dcrtBits / relinWindow) + 1;
```

Comment:
a few parentheses would improve the readability. also std::floor doesn't make any sense here as this is a division of 2 unsigned integers (even one of them declared as "int").

Reply:
Ok with parentheses. Please removed floor as it is not needed (it is already implied as you said).